File: prosessering/src/test/java/no/nav/familie/prosessering/task/Task2.java
package no.nav.familie.prosessering.task;

import no.nav.familie.prosessering.AsyncTask;
import no.nav.familie.prosessering.TaskBeskrivelse;
import no.nav.familie.prosessering.domene.Task;
import org.springframework.stereotype.Service;

import java.util.concurrent.TimeUnit;

@Service
@TaskBeskrivelse(taskType = Task2.TASK_2, beskrivelse = "Dette er task 2")
public class Task2 implements AsyncTask {

    public static final String TASK_2= "task2";


    @Override
    public void doTask(Task task) {

        try {
            TimeUnit.MICROSECONDS.sleep(1);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onCompletion(Task task){

    }
}

File: prosessering/src/test/java/no/nav/familie/prosessering/task/Task2.kt
package no.nav.familie.prosessering.task

import no.nav.familie.prosessering.AsyncTask
import no.nav.familie.prosessering.TaskBeskrivelse
import no.nav.familie.prosessering.domene.Task
import org.springframework.stereotype.Service
import java.util.concurrent.TimeUnit

@Service @TaskBeskrivelse(tasktype = Task2.TASK_2, beskrivelse = "Dette er task 2")
class Task2 : AsyncTask {
    override fun doTask(task: Task) {
        try {
            TimeUnit.MICROSECONDS.sleep(1)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }

    override fun preCondition(task: Task) {
        super.preCondition(task)
    }

    override fun onCompletion(task: Task) {}

    companion object {
        const val TASK_2 = "task2"
    }
}
